src/sentry/search/snuba/executors.py

```diff
@@ -472,6 +478,323 @@ def query(
+        "date": "events.last_seen",
+        "freq": "times_seen",
+        "new": "events.first_seen",
+        "priority": "priority",
+    }
+
+    aggregation_defs = {
+        "times_seen": ["count()", ""],
+        "events.first_seen": ["multiply(toUInt64(min(events.timestamp)), 1000)", ""],
+        "events.last_seen": ["multiply(toUInt64(max(events.timestamp)), 1000)", ""],
+        "priority": ["toUInt64(plus(multiply(log(times_seen), 600), `events.last_seen`))", ""],
+        "total": ["uniq", ISSUE_FIELD_NAME],
+    }
+
+    @property
+    @abstractmethod
+    def dataset(self):
+        return snuba.Dataset.Groups
+
+    def _transform_converted_filter(
+        self, search_filter, converted_filter, project_ids=None, environment_ids=None
+    ):
+        converted_name = search_filter.key.name
+
+        special_date_names = ["active_at", "first_seen", "last_seen"]
+        if search_filter.key.name in special_date_names:
+            # Need to get '2018-02-06T03:35:54' format out of 1517888878000 format
+            datetime_value = datetime.datetime.fromtimestamp(converted_filter[2] / 1000)
+            datetime_value = datetime_value.replace(microsecond=0).isoformat().replace("+00:00", "")
+            converted_filter[2] = datetime_value
+
+        # TODO: There is a better way to do this...the issue is that the table/alias is forked on environments, which can't be used in constrain_column_to_dataset
+        if search_filter.key.name in ["first_seen", "last_seen"]:  # , "first_release"]:
+            if environment_ids is not None:
+                table_alias = "events."
+                # return None, None #??? remove as a filter, since we are going to pre/post filter?
+            else:
+                table_alias = "groups."
+
+            if isinstance(converted_filter[0], list):
+                converted_filter[0][1][0] = table_alias + converted_filter[0][1][0]
+                converted_name = converted_filter[0][1][0]
+            else:
+                converted_filter[0] = table_alias + converted_filter[0]
+                converted_name = converted_filter[0]
+
+        if search_filter.key.name == "first_release":
+            # The filter's value will be the release's "version". Snuba only knows about ID. So we convert version to id here.
+            release = Release.objects.filter(
+                version=converted_filter[2],
+                organization_id=Project.objects.get(id=project_ids[0]).organization_id,
+            )
+            if not release:
+                # TODO: This means there will be no results and we do not need to run this query!
+                # right now it could lead to undesired results...if -1 is a real release id
+                # -1 is a number I'm hoping will never be a real release id
+                converted_filter[2] = -1
```

Comment:
This is a hint that this type of fitlering should not be done here, but in the query method while you are still on time for not querying Snuba.

Reply:
I moved this into the query function so we can avoid doing snuba queries if we don't find a match.